pkgs/development/libraries/qt-6/modules/qtwebengine.nix

```diff
@@ -118,12 +157,25 @@ qtModule {
 
     substituteInPlace src/3rdparty/chromium/ui/events/ozone/layout/xkb/xkb_keyboard_layout_engine.cc \
       --replace "/usr/share/X11/xkb" "${xkeyboard_config}/share/X11/xkb"
-
-    # Patch library paths in sources
+  ''
+  # Patch library paths in sources
+  + ''
     substituteInPlace src/core/web_engine_library_info.cpp \
       --replace "QLibraryInfo::path(QLibraryInfo::DataPath)" "\"$out\"" \
       --replace "QLibraryInfo::path(QLibraryInfo::TranslationsPath)" "\"$out/translations\"" \
       --replace "QLibraryInfo::path(QLibraryInfo::LibraryExecutablesPath)" "\"$out/libexec\""
+  ''
+  + lib.optionalString stdenv.isDarwin ''
+    substituteInPlace configure.cmake \
+      --replace "AppleClang" "Clang"
+    substituteInPlace cmake/Functions.cmake \
+      --replace "/usr/bin/xcrun" "${xcbuild}/bin/xcrun"
+    substituteInPlace src/3rdparty/chromium/build/config/mac/BUILD.gn \
+      --replace "\$mac_deployment_target" "${stdenv.targetPlatform.darwinSdkVersion}"
```

Comment:
This should be replaced by `"-DCMAKE_OSX_DEPLOYMENT_TARGET=${stdenv.targetPlatform.darwinSdkVersion}"` in `cmakeFlags`.